### PR TITLE
Invalidate onCameraMove as part of tracking animation

### PR DESCRIPTION
**Pull request by tobrun**
Closes https://github.com/mapbox/mapbox-plugins-android/issues/391

cc @danesfeder @LukasPaczos

### CODE REVIEW

File: plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java [hunk 1]
```diff
@@ -52,7 +52,7 @@
  *
  * @since 0.1.0
  */
-public final class LocationLayerPlugin implements LifecycleObserver {
+public final class LocationLayerPlugin implements LifecycleObserver, OnCameraMoveInvalidateListener {
```
Comment (tobrun) [inline plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: would be nicer to have an anonymous inner class so we don't add unnecessary public methods to the LocationLayerPlugin api

Comment (cammace) [reply plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: `OnCameraMoveInvalidateListener` is package private, I don't see the issue here?

Comment (LukasPaczos) [reply plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayerPlugin.java]: @cammace When implementing an interface we are unnecessary exposing `LocationLayerPlugin#onInvalidateCameraMove`. Making it an anonymous inner class hides that method.

**Comment by cammace** [comment]
I've moved the implementation into a private variable and fixed it not getting invoked inside `setBearing`. For the optimizations with onCameraMoved, I have ticketed it out and will work on it in a separate PR https://github.com/mapbox/mapbox-plugins-android/issues/413

**Pull request merged**